server/src/main/java/org/elasticsearch/index/mapper/DotExpandingXContentParser.java

```diff
@@ -171,37 +167,6 @@ private static void throwOnBlankOrEmptyPart(String field, String part) {
         protected XContentParser delegate() {
             return parsers.peek();
         }
-
-        @Override
-        public Map<String, Object> map() throws IOException {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public Map<String, Object> mapOrdered() throws IOException {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public Map<String, String> mapStrings() throws IOException {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public <T> Map<String, T> map(Supplier<Map<String, T>> mapFactory, CheckedFunction<XContentParser, T, IOException> mapValueParser)
-            throws IOException {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public List<Object> list() throws IOException {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public List<Object> listOrderedMap() throws IOException {
-            throw new UnsupportedOperationException();
-        }
```

Comment:
The actual implementations here will be those in AbstractXContentParser which just delegates things to nextToken(), currentToken(), and the various value methods.  So I think it should be safe.  It would be good to add some tests that check it does actually do what we expect though?

Reply:
> It would be good to add some tests that check it does actually do what we expect though?

Right, that's a good idea. I tried through the percolator tests but it became difficult, yet we can just expand the unit test for the parser itself. Will do that.

Reply:
i added tests, and figured that when expanding dots is enabled, the behaviour is unexpected in that we don't expand dots at all (we'd need to override an additional nextFieldName method). Given that we don't need the functionality when expanding dots, as we can safely assume that when parsing ordinary documents we never call map or list, I made all methods conditional of whether dots expansion is enabled or not.